src/app/admins/admins.pipe.ts

```diff
@@ -2,28 +2,22 @@ import { Pipe, PipeTransform } from "@angular/core";
 import { Store } from "@ngxs/store";
 import { IAdmin } from "./admins.interfaces";
 import { AdminsState } from "./state/admins.state";
+import { AuthService } from "../core/auth/auth.service";
+import { switchMap } from "rxjs/operators";
 
 @Pipe({
   name: "AdminName"
 })
 export class AdminsPipe implements PipeTransform {
-  private admins: IAdmin[] = this.store.selectSnapshot(AdminsState).items;
-
-  constructor(private store: Store) {}
+  constructor(private authService: AuthService) {}
 
   transform(username: string, format?: string): string {
-    if (username && this.admins) {
-      const admin: IAdmin = this.admins.find(
-        (a: IAdmin) => a.username === username
-      );
-
-      return admin
-        ? format === "email"
-          ? admin.email
-          : admin.fullName
-        : username;
+    if (username) {
```

Comment:
this will break the pipe usage in templates and tests.

Reply:
I think this always return the logged in User name, changing this pipe is not a solution.